### ISSUE TITLE
[Bug]: Usage of nodes that do not have a "required" dict results in no right click menu for the node

### ISSUE THREAD

**Issue by TashaSkyUp** [Comfy-Org/ComfyUI_frontend#510]
### Frontend Version

No clue as you have not revealed how to check that, and not everyone is a JS/TS/FRONTEND developer 

### Expected Behavior

there should still be a right click menu for nodes that don't have a required input, furthermore nodes with only optional inputs should be fully supported.

### Actual Behavior

an error is raised and execution of the thread is blocked here:
![image](https://github.com/user-attachments/assets/fbc9662e-7538-4efc-bd3e-0c8d875ae56a)

### Steps to Reproduce

have a node with only optional inputs and try to right click it.

### Debug Logs

```powershell
None in the server
```

### Browser Logs

Uncaught TypeError: Cannot read properties of undefined (reading 'INT')
    at ComfyNode.getConfig (widgetInputs.ts:382:22)
    at nodeType.getExtraMenuOptions (widgetInputs.ts:702:38)
    at nodeType.getExtraMenuOptions (McBoaty_v5.js:561:43)
    at nodeType.getExtraMenuOptions (node_fixer.js:17:21)
    at nodeType.getExtraMenuOptions (McBoaty_v4.js:146:43)
    at nodeType.getExtraMenuOptions (use_everywhere.js:83:34)
    at node.getExtraMenuOptions (comfyui-manager.js:1348:36)
    at nodeType.getExtraMenuOptions (menu_queue_node.js:37:55)
    at nodeType.getExtraMenuOptions (workflows.ts:241:21)
    at LGraphCanvas.getNodeMenuOptions (litegraph.es.js:9949:25)

### Other

I would fix this for my own fork but with this new crazy front end system you guys have made everything seems to be compiled
![image](https://github.com/user-attachments/assets/f8fc482e-6499-41df-8fa0-7b3a007c9ee8)
This points to a file that doesn't exist on the non-running server, so effectively I have no way of knowing how to make changes.
You may have made making changes easier for front end developers but you may have made it impossible for the python devs.

![image](https://github.com/user-attachments/assets/3ec7cdb4-5db3-4de6-a3a5-5583ceb0b667)

are these files in web/assets compiled? will their names change? what do I actually edit to make changes now??

**Comment by huchenlei** [comment]
![image](https://github.com/user-attachments/assets/bd22cc8c-3f9f-4ea7-a72c-22c718d7ecf3)
![image](https://github.com/user-attachments/assets/ea72a913-c0fa-4ce6-a71d-959ea5cf3bb2)

**Comment by huchenlei** [comment]
You can read https://github.com/Comfy-Org/ComfyUI_frontend#development for frontend dev setup guide.

Briefly:
1. Setup node and npm
2. Checkout this repo
3. Run `npm i`
4. Start ComfyUI at localhost:8188
5. Run `npm run dev` to start dev server

**Comment by huchenlei** [comment]
Bug confirmed reproducible.

**Comment by TashaSkyUp** [comment]
@huchenlei Thank you

![image](https://github.com/user-attachments/assets/bdffcdfb-d743-47f7-af13-8f72c76e07b4)


RANT
 I mean it doesn't help but it explains that ill have to spend countless hours learning front end to make simple changes now. This new front end setup I'm sure is great for people familiar with the tech who already have their environment setup etc etc etc. But I fail to see any advantage whatsoever vrs the way it was. seems like just buzzwords "Where using the latest design patterns!" Sounds good right? The reality is you guys just dumped lots of work on python devs and **completely cut out** quite a few of them.
/RANT

from the link:

> 
> Dev Server
> Note: The dev server will NOT load any extension from the ComfyUI server. Only core extensions will be loaded.
> 
> Run npm install to install the necessary packages
> Start local ComfyUI backend at localhost:8188
> Run npm run dev to start the dev server

how is this useful? Why would I want to **NOT** see if the nodes I'm developing actually work?

I mean yes I've done this npm building nonsense plenty before but it always just made things harder.
 
Is there something somewhere that lists the actual **concrete** advantages of the change? I actually would love to see the purpose, it will make the learning easier and make it at least seem more worth it.

Also **thank you** for #514 unfortunately I still have hundreds of nodes each of which I may have made custom changes to the JS and my productivity is going to be zero for the next week or so. Stressssed out.

**Comment by huchenlei** [comment]
If your custom node has JS code, you might want to follow https://github.com/Comfy-Org/ComfyUI_frontend#deploy to build and copy the compiled code to ComfyUI web root dir. If you don't want to mess up with your ComfyUI checkout, you can specify a temporary web root with `--front-end-root web_custom_versions/main/dev`.

The frontend migration change was noticed in https://github.com/comfyanonymous/ComfyUI/issues/4169.

**Comment by TashaSkyUp** [comment]
its because I made custom changes to multiple js files that where not meant to be changed (because I couldn't find another way (because im not a front end dev)). Not only did I make custom changes to js that wasn't meant to be changed I made changes to main.py, server.py, execute.py as well. ever since comfyui was released. I had been managing to keep my version current enough but these changes just destroyed my ability to re-integrate everything. That and I'm sick of dreading merging every few months.

largely the problems stem from I think most people only want comfy to be an image generation software.

But the way I use it its closer to an IDE.

I'm going to have to rework my workflows find work arounds that stay within the expected supported framework and the thing I dislike doing the most, _ask for help_.
